### PR TITLE
Refactor/categories to enums

### PR DESCRIPTION
**Pull request by abdokin**

### CODE REVIEW

**Comment by abdokin** [comment]
[For the custom questions category and test category, transfer them from table to enum](https://trello.com/c/hbVQ6LFQ/48-for-the-custom-questions-category-and-test-category-transfer-them-from-table-to-enum)

File: app/queries/test_query.rb [hunk 1]
```diff
@@ -16,10 +16,10 @@ def filter_by_search_query(search_query)
     @relation = relation.filter_by_search_query(search_query)
   end
 
-  def filter_by_categories(category_ids)
-    return relation if category_ids.blank?
+  def filter_by_categories(categories: nil)
```
Comment (itsAndL) [inline app/queries/test_query.rb:19]: i see here that you don't use this param "categories"

File: app/components/filter_component.rb [hunk 2]
```diff
@@ -101,17 +99,18 @@ def highlight_option(value, icon, label, count)
...
 
-  def test_focus_section(categories)
+  def test_focus_section
     {
       title: t('.test_focus'),
-      options: categories.order(:title).map.with_index(1) do |test_type, index|
-        count = @record_counts[:category_counts][test_type.title] || 0
+      options: Test.categories.keys.map.with_index(1) do |category, index|
+        count = @record_counts[:category_counts][category] || 0
+        # name, value, checked, id, label
```
Comment (itsAndL) [inline app/components/filter_component.rb:107]: This comment for what?

File: app/components/filter_component.rb [hunk 3]
```diff
@@ -183,17 +182,17 @@ def question_type_section
...
       title: t('.question_category'),
-      options: categories.order(:title).map.with_index(1) do |question_type, index|
-        count = @record_counts[:category_counts][question_type.title] || 0
+      options: CustomQuestion.categories.keys.map.with_index(1) do |category, index|
+        count = @record_counts[:category_counts][category] || 0
         checkbox_option(
           'question_category[]',
-          question_type.id,
-          params[:question_category]&.include?(question_type.id.to_s),
+          category,
```
Comment (itsAndL) [inline app/components/filter_component.rb:192]: in test_category_section you called .to_s (category.to_s), which one to use?

**Review by itsAndL — CHANGES_REQUESTED** [review]

File: db/schema.rb [hunk 4]
```diff
@@ -133,7 +133,7 @@
   end
 
   create_table 'custom_question_categories', force: :cascade do |t|
-    t.string 'title'
+    t.string 'title', null: false
     t.datetime 'created_at', null: false
     t.datetime 'updated_at', null: false
   end
```
Comment (itsAndL) [inline db/schema.rb:139]: this table needs to be drop, the same for test_categories
+ when i rollback i need to get the same table

File: db/schema.rb [hunk 5]
```diff
@@ -160,11 +160,11 @@
     t.integer 'duration_seconds', default: 0
     t.string 'type'
     t.integer 'position'
-    t.bigint 'custom_question_category_id', null: false
     t.datetime 'created_at', null: false
     t.datetime 'updated_at', null: false
     t.bigint 'business_id'
     t.integer 'language', default: 0
+    t.bigint 'custom_question_category_id'
```
Comment (itsAndL) [inline db/schema.rb:167]: the reference to be remove the same for test_categories, instead a "category" column of type integer, also when i rollback get the same reference

**Review by itsAndL — CHANGES_REQUESTED** [review]

**Review by itsAndL — APPROVED** [review]

**Pull request merged**